main-command/src/main/scala/sbt/internal/client/NetworkClient.scala

```diff
@@ -328,13 +328,39 @@ class NetworkClient(
             term.isSupershellEnabled
           ).mkString(",")
 
-        val cmd = arguments.sbtLaunchJar match {
+        val (cmd: List[String], envs: Map[String, String]) = arguments.sbtLaunchJar match {
           case Some(lj) =>
-            List("java") ++ arguments.sbtArguments ++
-              List("-jar", lj, DashDashDetachStdio, DashDashServer)
+            def loadConfigFile(file: File): List[String] =
+              IO.readLines(file)
+                .map(_.trim)
+                .filter(_.nonEmpty)
+                .filterNot(_.startsWith("#"))
+
+            val jvmOpts =
+              loadConfigFile(file(".jvmopts"))
+            val sbtOpts =
+              loadConfigFile(file("/etc/sbt/sbtopts")) ++ loadConfigFile(file(".sbtopts"))
+
+            val javaOpts = jvmOpts ++ sbtOpts.collect {
+              case jvmOpt if jvmOpt.startsWith("-X") || jvmOpt.startsWith("-D") => jvmOpt
+              case prefixedJvmOpt if prefixedJvmOpt.startsWith("-J-X")          => prefixedJvmOpt.drop(2)
```

Comment:
this will silently ignore options like `-debug` or `--supershell` which are valid `.sbtopts` options, but since some of them are handled in the shell script itself, I figured out I would focus on JVM options first.